pkgs/stdenv/generic/default.nix

```diff
@@ -319,17 +331,15 @@ let
           userHook = config.stdenv.userHook or null;
           __ignoreNulls = true;
 
-          # Inputs built by the cross compiler.
-          buildInputs = buildInputs';
-          propagatedBuildInputs = propagatedBuildInputs';
-          # Inputs built by the usual native compiler.
-          nativeBuildInputs = nativeBuildInputs'
-            ++ lib.optional
-                (hostPlatform.isCygwin
-                  || (crossConfig != null && lib.hasSuffix "mingw32" crossConfig))
-                ../../build-support/setup-hooks/win-dll-link.sh
-            ;
-          propagatedNativeBuildInputs = propagatedNativeBuildInputs';
+          preNativeBuildInputs = lib.elemAt dependencies' 0;
```

Comment:
I haven't looked at the details (sorry (!)), but organizing data in this way, instead of using attribute sets, seems awful.